pkg/datastore/postgres.go

```diff
@@ -274,66 +241,37 @@ func (ds *Datastore) Update(ctx context.Context, ve VersionedJSONEntity) error {
 
 // Get the entity for given id
 // returns NotFoundError if no entity can be found
-func (ds *Datastore) Get(ctx context.Context, id string, ve VersionedJSONEntity) error {
-	jsonField := ve.JSONField()
-	tableName := ve.TableName()
-	_, ok := ds.types[jsonField]
-	if !ok {
-		return fmt.Errorf("type:%s is not registered", jsonField)
-	}
+func (ds *datastore[E]) Get(ctx context.Context, id string) (E, error) {
+	ds.log.Debug("get", zap.String("entity", ds.jsonField), zap.String("id", id))
+	var zero E
 	q := ds.sb.Select(
-		jsonField,
+		ds.jsonField,
 	).From(
-		tableName,
+		ds.tableName,
 	).Where(squirrel.Eq{
 		"id": id,
 	})
 
-	sql, _, _ := q.ToSql()
-	ds.log.Info("get", zap.String("entity", jsonField), zap.String("sql", sql), zap.String("id", id))
-	rows, err := q.QueryContext(ctx)
+	row := q.QueryRowContext(ctx)
+	e := new(E)
+	err := row.Scan(e)
 	if err != nil {
-		return err
-	}
-	defer func() {
-		_ = rows.Close()
-		_ = rows.Err()
-	}()
-	if rows.Next() {
-		return rows.Scan(ve)
+		return zero, NewNotFoundError(fmt.Sprintf("entity of type:%s with id:%s not found %v", ds.jsonField, id, err))
```

Comment:
Does this properly distinguish between not found and a database error? 

For a database error it's interesting for the caller to get the error, in case of not found I think an error like "tenant with id:fjkdf not found" and that's it.

Reply:
done

Reply:
done